### PR TITLE
Fix basic-deployment kuttl assert

### PR DESCRIPTION
**Pull request by ashu-011**
Currently the way kuttl test is written , it will still pass if no status code is returned as script doesn't have proper exit mechanism to handle that . This PR fixes that and introduce proper error message . Also fixes the command to run in required namespace .

### CODE REVIEW

File: tests/kuttl/common/assert-sample-deployment.yaml [hunk 1]
```diff
@@ -20,7 +20,7 @@ commands:
       RETURN_CODE=0
       PUBLIC_URL=$(oc get -n $NAMESPACE horizon horizon -o jsonpath='{.status.endpoint}')
```
Comment (bshephar) [inline tests/kuttl/common/assert-sample-deployment.yaml:21]: So this one successfully gets the endpoint from the statue:
```
logger.go:42: 11:15:54 | basic-deployment/1-deploy-horizon | ++ oc get -n horizon-kuttl-tests horizon horizon -o 'jsonpath={.status.endpoint}'
    logger.go:42: 11:15:54 | basic-deployment/1-deploy-horizon | + PUBLIC_URL=http://horizon.horizon-kuttl-tests.svc/

```

This one fails:
```
    logger.go:42: 11:15:54 | basic-deployment/1-deploy-horizon | ++ oc exec -it deployment/horizon -- curl --silent --output /dev/stderr --head --write-out '%{http_code}' 'http://horizon.horizon-kuttl-tests.svc/dashboard/auth/login/?next=/dashboard/' -k
    logger.go:42: 11:15:54 | basic-deployment/1-deploy-horizon | Error from server (NotFound): namespaces "ci-op-tt7vckqd" not found
```

It just needs the `-n $NAMESPACE` in the command.

**Review by bshephar — CHANGES_REQUESTED** [review]

File: tests/kuttl/common/assert-sample-deployment.yaml [hunk 2]
```diff
@@ -20,7 +20,7 @@ commands:
...
       STATUSCODE=$(oc exec -it deployment/horizon -- curl --silent --output /dev/stderr --head --write-out "%{http_code}" "$PUBLIC_URL/dashboard/auth/login/?next=/dashboard/" -k)
```
Comment (bshephar) [inline tests/kuttl/common/assert-sample-deployment.yaml:22]: This one fails:
```
    logger.go:42: 11:15:54 | basic-deployment/1-deploy-horizon | ++ oc exec -it deployment/horizon -- curl --silent --output /dev/stderr --head --write-out '%{http_code}' 'http://horizon.horizon-kuttl-tests.svc/dashboard/auth/login/?next=/dashboard/' -k
    logger.go:42: 11:15:54 | basic-deployment/1-deploy-horizon | Error from server (NotFound): namespaces "ci-op-tt7vckqd" not found
```

It just needs the `-n $NAMESPACE` in the command.

**Review by bshephar — CHANGES_REQUESTED** [review]

File: tests/kuttl/common/assert-sample-deployment.yaml (hunk 1 above)
Comment (bshephar) [reply tests/kuttl/common/assert-sample-deployment.yaml:21]: Sorry, no change required for this line. Just highlighting that this command works and it contains the `-n $NAMESPACE` while the next command doesn't and is missing the `-n $NAMESPACE`

File: tests/kuttl/common/assert-sample-deployment.yaml [hunk 3]
```diff
@@ -19,8 +19,8 @@ commands:
       set -x
       RETURN_CODE=0
       PUBLIC_URL=$(oc get -n $NAMESPACE horizon horizon -o jsonpath='{.status.endpoint}')
-      STATUSCODE=$(oc exec -it deployment/horizon -- curl --silent --output /dev/stderr --head --write-out "%{http_code}" "$PUBLIC_URL/dashboard/auth/login/?next=/dashboard/" -k)
-      if test $STATUSCODE -ne 200; then
+      STATUSCODE=$(oc -n $NAMESPACE exec -it deployment/horizon -- curl --silent --output /dev/stderr --head --write-out "%{http_code}" "$PUBLIC_URL/dashboard/auth/login/?next=/dashboard/" -k)
```
Comment (bshephar) [inline tests/kuttl/common/assert-sample-deployment.yaml:22]: lgtm now. Let's see how kuttl likes it.

File: tests/kuttl/common/assert-sample-deployment.yaml [hunk 4]
```diff
@@ -19,8 +19,8 @@ commands:
...
+      if [ -z ${STATUSCODE} ] || test $STATUSCODE -ne 200; then
           RETURN_CODE=1
           echo "${PUBLIC_URL} status code expected is 200 but was ${STATUSCODE}"
```
Comment (odyssey4me) [inline tests/kuttl/common/assert-sample-deployment.yaml:25]: If there is no status code, then isn't that a problem we care about and it should still exit 1 but with a different error message?

File: tests/kuttl/common/assert-sample-deployment.yaml (hunk 3 above)
Comment (ashu-011) [reply tests/kuttl/common/assert-sample-deployment.yaml:22]: looks good as job ran successfully , thank you

**Review by bshephar — APPROVED** [review]

**Pull request merged**